src/main/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImpl.java

```diff
@@ -98,12 +98,14 @@ public void extractAchPayments() {
         achFilePrefix = MessageFormat.format(achFilePrefix, new Object[] { null });
     
         String filename = getOutputFile(achFilePrefix, processDate);
-        LOG.debug("MOD: extractAchPayments() filename = " + filename);
 
         /** 
         * MOD: This is the only section in the method that is changed.  This mod calls a new method that bundles 
         * ACHs into single disbursements if the flag to do so is turned on.
+        * Modified the SimpleDateFormat to have milliseconds to prevent files being overwritten
         */
+        sdf = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss-SSS", Locale.US);
```

Comment:
I don't think this particular date format needs updating, because this line of code should be unreachable when ISO 20022 formatting is enabled.  (The `if` block above with the `shouldUseIso20022Format()` check should force the method to return early.) Also, the downstream code might be ignoring or replacing this format anyway.

I believe the correct way to modify the filename for the legacy Check/ACH files is to change the date formatter in the `getOutputFile()` method. That method is in base code but can be overridden by this subclass. (Note that we're still generating legacy KFS-proprietary Check files in addition to ISO Check files, so such an override will be needed until we turn off the KFS-proprietary Check file handling in a future sprint.)